tools/jit/gen_jit_dispatch.py

```diff
@@ -85,127 +84,123 @@ def is_jit_op(decl):
 
 
 def gen_jit_dispatch(declarations, out):
-    # We need to add methods implemented manually in TensorImpl
-    tensor_impl_methods = [{
-        'name': name,
-        'api_name': name,
-        'method_of': ['Tensor'],
-        'arguments': [{'name': 'self', 'simple_type': 'Tensor'}],
-    } for name in ['sizes', 'strides', 'dim']]
-    aten_decls = load_aten_declarations(declarations) + tensor_impl_methods
-    jit_decls = [d for d in aten_decls if is_jit_op(d)]
+    ops = {}
 
     def is_tensor_arg(arg):
         return arg['simple_type'] in {'Tensor', 'TensorList'}
 
-    ops = {}
-    names = set()
-    for decl in jit_decls:
+    def get_invocation(decl, args):
+        if 'namespace' in decl['method_of']:
+            return CALL_NAMESPACE.substitute(name=decl['name'], args=args)
+        else:
+            return CALL_METHOD.substitute(name=decl['name'], first=args[0], args=args[1:])
+
+    def emit_decl_variant(decl, is_real_input, has_tensorlist):
+        kw_assignments = []
+        attr_names = []
+        pos_assignments = []
+        arguments = []
+
+        if has_tensorlist:
+            kw_assignments.append('size_t varargs_length = node->inputs().size();')
+            # arguments look like: [tensor list], other, static, inputs
+            # we use peek(<i>, static_inputs) to read the non-vararg inputs
+            # from the end of the stack
+            static_inputs = sum(is_real_input) - 1
+            num_dynamic_inputs = 'varargs_length'
+        else:
+            static_inputs = sum(is_real_input)
+            num_dynamic_inputs = static_inputs
+
+        real_inputs = count()
+        for i, arg in enumerate(decl['arguments']):
+            # XXX: we currently support only TensorList ops that have a TensorList as
+            # the first argument, that is then followed by a number of positional args.
+            if arg['simple_type'] == 'TensorList':
+                arguments.append('peekSlice(stack, 0, varargs_length - {}, varargs_length)'.format(static_inputs))
```

Comment:
This no longer moves from the stack

Reply:
It called `last...` before which also did not move off the stack. I think this is fine for now, because we can't move the TensorList in to ATen op anyway, so we wouldn't be cutting down on lifetimes in either case. Regardless of if something is moved, `drop(` is going to delete the stack entry and free the tensor if it is still there.